features/apimgt/org.wso2.carbon.apimgt.admin.feature/src/main/resources/admin_new/source/src/app/components/Throttling/Custom/AddEdit.jsx

```diff
@@ -100,8 +120,22 @@ function AddEdit(props) {
     const [validationError, setValidationError] = useState([]);
     const [editMode, setIsEditMode] = useState(false);
     const restApi = new API();
+    const { policyId } = props.match.params;
 
     useEffect(() => {
+        if (policyId !== 'create') {
+            setIsEditMode(true);
+            restApi.customPolicyGet(policyId).then((result) => {
+                const formattedSiddhiQuery = sqlFormatter.format(result.body.siddhiQuery);
+                const editState = {
+                    policyName: result.body.policyName,
+                    description: result.body.description,
+                    keyTemplate: result.body.keyTemplate,
+                    siddhiQuery: formattedSiddhiQuery,
+                };
+                dispatch(editState);
```

Comment:
```javascript
                dispatch(editState);
```
Though this `dispatch` call does the job here, It's better to follow the action dispatching methodology which is described in [Flux pattern](https://facebook.github.io/flux/docs/in-depth-overview#actions).
In the Flux pattern Object that we pass into the `dispatcher` called an action, Typically there are two elements in an action type and payload, and we derive the next state by switching the executing by the action type and incoming event's payload.

This pattern can be seen in [React documentation examples](https://reactjs.org/docs/hooks-reference.html#lazy-initialization) as well.

Shall we follow a similar pattern here as well?

Reply:
Fixed with `4e14418`